u8timeseries/timeseries.py

```diff
@@ -398,17 +398,19 @@ def from_times_and_values(times: pd.DatetimeIndex,
 
         return TimeSeries(series, series_lo, series_hi)
 
-    def plot(self, *args, plot_ci=True, **kwargs):
+    def plot(self, *args, plot_ci=True, new_plot=False, **kwargs):
         """
         Currently this is just a wrapper around pd.Series.plot()
         """
         # temporary work-around for the pandas.plot issue
         # errors = self._combine_or_none(self._confidence_lo, self._confidence_hi,
         #                                lambda x, y: np.vstack([x.values, y.values]))
         # self._series.plot(yerr=errors, *args, **kwargs)
-        fig, ax = plt.subplots()
-        ax.plot_date(self.time_index(), self.values(), marker='', linestyle='-', *args, **kwargs)
-        fig.autofmt_xdate()
+        if new_plot:
+            fig, ax = plt.subplots()
+        else:
+            fig, ax = plt.gcf(), plt.gca()
+        self._series.plot()
```

Comment:
So now that you do it this way, you never use the `fig` and `ax` variables, so I don't think the `new_plot` flag will work when you set it to True. What you need is probably something like:
```
fig = (plt.figure() if new_plot else plt.gcf())
self._series.plot(figure=fig)
```

Reply:
done